include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -737,6 +737,18 @@ def RefSendOp: FIRRTLExprOp<"ref.send", [RefResultTypeConstraint<"base", "result
   let assemblyFormat = "$base attr-dict `:` qualified(type($base))";
 }
 
+def RefSendInternalPathOp: FIRRTLOp<"ref.send.internalPath", [Pure, HasCustomSSAName]>{
+  let summary = "FIRRTL Send an opaque internal path";
+  let description = [{
+    Endpoint of a remote reference. Send an opaque internal path
+    through a reference to be read from the firrtl.ref.resolve op.
+  }];
+  let arguments = (ins StrAttr:$path);
+  let results = (outs RefType:$result);
+
+  let assemblyFormat = "$path attr-dict `:` qualified(type($result))";
+}
```

Comment:
An alternative to adding this op would be to just use an `sv::VerbatimExpr` and feed that into a `RefSend`. Noting that I think both that and the approach here are _extremely_ janky as they are encoding an arbitrary _Verilog_ XMR string that will only be valid if the pass and optimization stars align.

Reply:
More accurately, I think this is really just `firrtl::VerbatimExprOp` and we don't need to create a new op for this. Just changing the creation of the `ref.send.internalPath` to use a `firrtl.verbatim.expr` and `firrtl.ref.send` will do the right thing, but with a dummy wire. This PR can learn about `firrtl.verbatim.expr` and inline that like it does for `ref.send.internalPath` and we can save an op with duplicated semantics.

Reply:
Right, we can use a `verbatim.expr` + `ref.send`. 
What if we only use `verbatim.expr` to generate a `RefType` value ? How about we allow `verbatim.expr` to generate `RefType`. This will be exact replacement of the proposed op above. This can then be guaranteed to be removed by `LowerXMR`.

Reply:
Having verbatimexpr generate reftype seems great to me.

Reply:
Clarifying my logic: a `verbatim.expr` is, my definition, is a verbatim expression and can thereby return any type. I agree that there's not a lot of utility in bouncing this through a `RefSend`. That said, it may be easier to do the bouncing because things like `LowerXMR` can then be assured that each `RefReceive` pairs with a `RefSend`. If using a `verbatim.expr`, then there are two types of endpoints.